File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/users/UsersSetPhotoRequest.java
package com.github.seratch.jslack.api.methods.request.users;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

import java.io.File;

/**
 * Set the user profile photo
 */
public class UsersSetPhotoRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `users.profile:write`
     */
    private String token;

    /**
     * File contents via `multipart/form-data`.
     */
    private File image;
    private byte[] imageData;

    /**
     * X coordinate of top-left corner of crop box
     */
    private Integer cropX;

    /**
     * Y coordinate of top-left corner of crop box
     */
    private Integer cropY;

    /**
     * Width/height of crop box (always square)
     */
    private Integer cropW;

    UsersSetPhotoRequest(String token, File image, byte[] imageData, Integer cropX, Integer cropY, Integer cropW) {
        this.token = token;
        this.image = image;
        this.imageData = imageData;
        this.cropX = cropX;
        this.cropY = cropY;
        this.cropW = cropW;
    }

    public static UsersSetPhotoRequestBuilder builder() {
        return new UsersSetPhotoRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public File getImage() {
        return this.image;
    }

    public byte[] getImageData() {
        return this.imageData;
    }

    public Integer getCropX() {
        return this.cropX;
    }

    public Integer getCropY() {
        return this.cropY;
    }

    public Integer getCropW() {
        return this.cropW;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setImage(File image) {
        this.image = image;
    }

    public void setImageData(byte[] imageData) {
        this.imageData = imageData;
    }

    public void setCropX(Integer cropX) {
        this.cropX = cropX;
    }

    public void setCropY(Integer cropY) {
        this.cropY = cropY;
    }

    public void setCropW(Integer cropW) {
        this.cropW = cropW;
    }

    public static class UsersSetPhotoRequestBuilder {
        private String token;
        private File image;
        private byte[] imageData;
        private Integer cropX;
        private Integer cropY;
        private Integer cropW;

        UsersSetPhotoRequestBuilder() {
        }

        public UsersSetPhotoRequest.UsersSetPhotoRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public UsersSetPhotoRequest.UsersSetPhotoRequestBuilder image(File image) {
            this.image = image;
            return this;
        }

        public UsersSetPhotoRequest.UsersSetPhotoRequestBuilder imageData(byte[] imageData) {
            this.imageData = imageData;
            return this;
        }

        public UsersSetPhotoRequest.UsersSetPhotoRequestBuilder cropX(Integer cropX) {
            this.cropX = cropX;
            return this;
        }

        public UsersSetPhotoRequest.UsersSetPhotoRequestBuilder cropY(Integer cropY) {
            this.cropY = cropY;
            return this;
        }

        public UsersSetPhotoRequest.UsersSetPhotoRequestBuilder cropW(Integer cropW) {
            this.cropW = cropW;
            return this;
        }

        public UsersSetPhotoRequest build() {
            return new UsersSetPhotoRequest(token, image, imageData, cropX, cropY, cropW);
        }

        public String toString() {
            return "UsersSetPhotoRequest.UsersSetPhotoRequestBuilder(token=" + this.token + ", image=" + this.image + ", imageData=" + java.util.Arrays.toString(this.imageData) + ", cropX=" + this.cropX + ", cropY=" + this.cropY + ", cropW=" + this.cropW + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/users/UsersSetPhotoRequest.kt
package work.socialhub.kslack.api.methods.request.users

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

/**
 * Set the user profile photo
 */
@JsExport
class UsersSetPhotoRequest(
    /** Authentication token. Requires scope: `users.profile:write` */
    override var token: String?,
    /** File contents via `multipart/form-data`. */
    var imageData: ByteArray,

    /** X coordinate of top-left corner of crop box */
    var cropX: Int?,
    /** Y coordinate of top-left corner of crop box */
    var cropY: Int?,
    /** Width/height of crop box (always square) */
    var cropW: Int?,
) : SlackApiRequest, FormRequest {
    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {

//            MultipartBody.Builder form = new MultipartBody.Builder();
//            if (ImageData() != null) {
//                RequestBody image = RequestBody.create(MediaType.parse("imageData/*"), ImageData);
//                form.addFormDataPart("image", "image", image);
//            } else if (Image() != null) {
//                RequestBody image = RequestBody.create(MediaType.parse("imageData/*"), Image);
//                form.addFormDataPart("image", "image", image);
//            }
//            it.addParam("crop_x", CropX);
//            it.addParam("crop_y", CropY);
//            it.addParam("crop_w", CropW);
//            return form;
        }
    }
}